### ISSUE TITLE
proposal: API change, add `elemType` parameter to `getelementptr` constructors

### ISSUE THREAD

**Issue by mewmew** [llir/llvm#112]
A quick heads up regarding an API change that will land before v0.3.

After working on fixing the result type computation of `getelementptr` these last few days, and reflecting briefly on future changes that are to land in the official LLVM project (in particular opaque pointer type), it seems like now is the time for us to introduce the `elemType` parameter to the `NewGetElementPtr` constructors for `gep` instructions and constant expressions.

Previously we've been able to infer `elemType` from `src` simply by running something to the effect of `src.Type().(*types.PointerType).ElemType`, but this would no longer be possible with opaque pointer types. Thus, lets anticipate the upstream change instead of being struck by it, and adjust our API accordingly.

In summary, the proposed API change is as follows.

[ir.NewGetElementPtr](https://godoc.org/github.com/llir/llvm/ir#NewGetElementPtr):
```diff
-func NewGetElementPtr(src value.Value, indices ...value.Value) *InstGetElementPtr
+func NewGetElementPtr(elemType types.Type, src value.Value, indices ...value.Value) *InstGetElementPtr
```

[ir.Block.NewGetElementPtr](https://godoc.org/github.com/llir/llvm/ir#Block.NewGetElementPtr)
```diff
-func (block *Block) NewGetElementPtr(src value.Value, indices ...value.Value) *InstGetElementPtr
+func (block *Block) NewGetElementPtr(elemType types.Type, src value.Value, indices ...value.Value) *InstGetElementPtr
```

[ir/constant.NewGetElementPtr](https://godoc.org/github.com/llir/llvm/ir/constant#NewGetElementPtr)
```diff
-func NewGetElementPtr(src Constant, indices ...Constant) *ExprGetElementPtr
+func NewGetElementPtr(elemType types.Type, src Constant, indices ...Constant) *ExprGetElementPtr
```

The change to user code should be fairly simple, and often fixable by search and replace. There is no intention to keep two versions of the API around.

cc: @pwaller

**Comment by mewmew** [comment]
An `elemType` would also be added to the `NewLoad` constructor for the same reason.

```go
// TODO: add elemType parameter to Newload, ElemType field to InstLoad and use
// ElemType in InstLoad.Type rather than recomputing the element type each time.
// This is also in preparation for the opaque pointer type of upcoming LLVM
// releases.
```

**Comment by mewmew** [comment]
This API change has been implemented as of e3e7aa1ad6c089949aba4a4aae14204e24d0acb8.
